include/dlaf/eigensolver/gen_to_std/mc/gen_to_std_L.h

```diff
@@ -80,21 +84,29 @@ void genToStd_L(Matrix<T, Device::CPU>& mat_a, Matrix<T, Device::CPU>& mat_l) {
         hpx::dataflow(executor_hp, unwrapping(tile::her2k<T, Device::CPU>), Lower, NoTrans, -1.0,
                       mat_a.read(jk), mat_l.read(jk), 1.0, mat_a(jj));
 
-        for (SizeType i = j + 1; i < m; ++i) {
-          const auto ik = LocalTileIndex{i, k};
-          const auto ij = LocalTileIndex{i, j};
-          hpx::dataflow(executor_normal, unwrapping(tile::gemm<T, Device::CPU>), NoTrans, ConjTrans,
-                        -1.0, mat_a.read(ik), mat_l.read(jk), 1.0, mat_a(ij));
-          hpx::dataflow(executor_normal, unwrapping(tile::gemm<T, Device::CPU>), NoTrans, ConjTrans,
-                        -1.0, mat_l.read(ik), mat_a.read(jk), 1.0, mat_a(ij));
+        if (j + 1 < m) {
+          LocalTileIndex istart(j + 1, 0);
+          LocalTileIndex iend(m, 1);
+          for (auto i : dlaf::common::iterate_range2d(istart, iend)) {
+            const auto ik = LocalTileIndex{i.row(), k};
+            const auto ij = LocalTileIndex{i.row(), j};
+            hpx::dataflow(executor_normal, unwrapping(tile::gemm<T, Device::CPU>), NoTrans, ConjTrans,
+                          -1.0, mat_a.read(ik), mat_l.read(jk), 1.0, mat_a(ij));
+            hpx::dataflow(executor_normal, unwrapping(tile::gemm<T, Device::CPU>), NoTrans, ConjTrans,
+                          -1.0, mat_l.read(ik), mat_a.read(jk), 1.0, mat_a(ij));
+          }
         }
       }
 
-      for (SizeType i = k + 1; i < m; ++i) {
-        // Working on panel...
-        const auto ik = LocalTileIndex{i, k};
-        hpx::dataflow(executor_hp, unwrapping(tile::hemm<T, Device::CPU>), Right, Lower, -0.5,
-                      mat_a.read(kk), mat_l.read(ik), 1.0, mat_a(ik));
+      if (k + 1 < m) {
+        LocalTileIndex istart(k + 1, 0);
+        LocalTileIndex iend(m, 1);
+        for (auto i : dlaf::common::iterate_range2d(istart, iend)) {
+          // Working on panel...
+          const auto ik = LocalTileIndex{i.row(), k};
```

Comment:
See my previous comment.

Moreover, I'd like to point out that you can re-use the same `iterate_range2d` iterator, since you have already used it. Just to make it more clear, I'll try to give share with you my idea.

In this algorithm you are going to work twice over the same panel, for which you added the comment `// Working on panel...`. IMHO, It would have been more clear to have

```cpp
LocalTileIndex ai_start(k + 1, k);
LocalTileSize ai_size(m - (k + 1), 1);
const auto ai_panel = dlaf::common::iterate_range2d(ai_start, ai_size);

// ...

for (const auto ik : ai_panel) {
}

// ...

for (const auto ik : ai_panel) {
}
```

and even the comment would look superfluous once the intentions are quite clear directly from `for (const auto ij : ai_panel)` since it already says that you want to iterate over the `ai_panel`.

@rasolca what do you think?

Reply:
I agree.